universal-application-tool-0.0.1/app/services/question/QuestionDefinition.java

```diff
@@ -0,0 +1,90 @@
+package services.question;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import java.util.Locale;
+
+/** Defines a single question. */
+public class QuestionDefinition {
+  private final String id;
+  private final String version;
+  private final String name;
+  private final String path;
+  private final String description;
+  private final ImmutableMap<Locale, String> questionText;
+  private final ImmutableMap<Locale, String> questionHelpText;
+  private final ImmutableSet<String> tags;
+
+  public QuestionDefinition(
+      String id,
+      String version,
+      String name,
+      String path,
+      String description,
+      ImmutableMap<Locale, String> questionText,
+      ImmutableMap<Locale, String> questionHelpText,
+      ImmutableSet<String> tags) {
+    this.id = id;
+    this.version = version;
+    this.name = name;
+    this.path = path;
+    this.description = description;
+    this.questionText = questionText;
+    this.questionHelpText = questionHelpText;
+    this.tags = tags;
```

Comment:
Suggestion (blocking): for anything that isn't explicitly nullable please include a null check.

```
import static com.google.common.base.Preconditions.checkNotNull;

this.id = checkNotNull(id);
```

Reply:
Done